### PR TITLE
Open runner PR on schemas release

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
This introduces a new GitHub Action in `release` yaml file that sends `repository dispatch` to runner whenever new version of eq-questionnaire-schemas is released, which then should automatically generate a new pull request. `repository dispatch` has specific event type so runner PR is getting triggered only on this specific occasion.

### How to review 
Check the code and try to trigger pull request by releasing new schemas version.

### CODE REVIEW

**Comment by petechd** [comment]
[Test load-schemas is up to date with latest version before allowing merge (S)](https://trello.com/c/rQfJ7j8n/3775-test-load-schemas-is-up-to-date-with-latest-version-before-allowing-merge-s)

**Comment by petechd** [comment]
[Test load-schemas is up to date with latest version before allowing merge (S)](https://trello.com/c/rQfJ7j8n/3775-test-load-schemas-is-up-to-date-with-latest-version-before-allowing-merge-s)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ on:
       - v*
 
 jobs:
+  repository-dispatch:
```
Comment (iwootten) [inline .github/workflows/release.yml:9]: This doesn't seem like a good name for the step/description. 

I'd prefer to use an existing action for this - e.g. https://github.com/marketplace/actions/repository-dispatch

Comment (petechd) [reply .github/workflows/release.yml]: Sorted, thanks!👍

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -6,10 +6,12 @@ on:
       - v*
 
 jobs:
-  repository-dispatch:
-    - name: Send repository dispatch
-      run: |
-        curl -XPOST -u "${{ secrets.USERNAME}}:${{secrets.TOKEN}}" -H "Accept: application/vnd.github.everest-preview+json"  -H "Content-Type: application/json" https://api.github.com/repos/ONSdigital/eq-questionnaire-runner/dispatches --data '{"event_type": "schemas_release"}'
+  name: Repository Dispatch
```
Comment (iwootten) [inline .github/workflows/release.yml:9]: I think the name of this step should be something like "Notify Runner".

Comment (petechd) [reply .github/workflows/release.yml]: Ok, no problem. Changed👍

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -6,6 +6,14 @@ on:
       - v*
 
 jobs:
+  notify-runner:
+    steps:
```
Comment (MebinAbraham) [inline .github/workflows/release.yml:10]: Needs `runs-on: ubuntu-latest`

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -6,6 +6,14 @@ on:
...
+      - name: Notify Runner
+        uses: peter-evans/repository-dispatch@v1.0.0
+        with:
+          token: ${{ secrets.TOKEN }}
```
Comment (MebinAbraham) [inline .github/workflows/release.yml:14]: As discussed, we need to make this TOKEN value using a service account.

Comment (MebinAbraham) [reply .github/workflows/release.yml:14]: We should make this specific and call it `REPO_ACCESS_TOKEN`

File: .github/workflows/release.yml (hunk 3 above)
Comment (petechd) [reply .github/workflows/release.yml:11]: Done, thanks!

File: .github/workflows/release.yml (hunk 4 above)
Comment (petechd) [reply .github/workflows/release.yml]: Someone needs to add this `REPO_ACCESS_TOKEN` now, done!

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -6,6 +6,14 @@ on:
       - v*
 
 jobs:
+  notify-runner:
```
Comment (MebinAbraham) [inline .github/workflows/release.yml:9]: Ideally this should depend on the `release` job passing before notifying.

Comment (petechd) [reply .github/workflows/release.yml]: Done, thanks👍

**Comment by MebinAbraham** [comment]
Small change to pass the TAG to runner and have also uploaded the `REPO_ACCESS_TOKEN` secret.

**Pull request merged**